### PR TITLE
cephadm/ceph-volume: do not use lvm binary in containers

### PR DESCRIPTION
**Pull request by guits**
ceph-volume should run pv/vg/lv commands in the host namespace rather than
running them inside the container in order to avoid lvm metadata corruption.

Fixes: https://tracker.ceph.com/issues/51592
Fixes: https://tracker.ceph.com/issues/52926

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -2420,6 +2420,7 @@ def get_container_mounts(ctx, fsid, daemon_type, daemon_id,
             if not os.path.exists(selinux_folder):
                 os.makedirs(selinux_folder, mode=0o755)
             mounts[selinux_folder] = '/sys/fs/selinux:ro'
+        mounts['/'] = '/rootfs:ro'
```
Comment (guits) [inline src/cephadm/cephadm:2423]: @sebastian-philipp is it enough? Are there other places where I should add this change?

Comment (sebastian-philipp) [reply src/cephadm/cephadm:2423]: We have to make sure `ceph ceph-volume` also properly mounts this. According to 

https://github.com/ceph/ceph/blob/d009895a6b2147a8f6f5bb32add778acdc42d598/src/cephadm/cephadm#L4990

this is already the case, so lgtm

**Review by sebastian-philipp — APPROVED** [review]

**Comment by guits** [comment]
jenkins test ceph-volume tox

**Comment by guits** [comment]
jenkins test ceph-volume all

**Comment by guits** [comment]
jenkins test ceph-volume tox

**Comment by guits** [comment]
jenkins test ceph-volume all

**Comment by guits** [comment]
jenkins test ceph-volume all

**Comment by guits** [comment]
jenkins test ceph-volume tox

**Comment by sebastian-philipp** [comment]
```
RuntimeError: cephadm exited with an error code: 1, stderr:Inferring config /var/lib/ceph/872584d6-3c00-11ec-8c28-001a4aab830c/mon.a/config
Non-zero exit code 1 from /bin/podman run --rm --ipc=host --stop-signal=SIGTERM --net=host --entrypoint /usr/sbin/ceph-volume --privileged >
/bin/podman: stderr --> passed data devices: 1 physical, 0 LVM
/bin/podman: stderr --> relative data size: 1.0
/bin/podman: stderr Running command: /usr/bin/ceph-authtool --gen-print-key
/bin/podman: stderr Running command: /usr/bin/ceph --cluster ceph --name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/ceph.ke>
/bin/podman: stderr Running command: nsenter --root=/rootfs --mount=/rootfs/proc/1/ns/mnt --ipc=/rootfs/proc/1/ns/ipc --net=/rootfs/proc/1/>
/bin/podman: stderr  stderr: Failed to find sysfs mount point
/bin/podman: stderr  stderr: dev/block/259:0/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/8:0/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/253:0/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/259:1/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/8:1/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/253:1/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/259:2/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/253:2/holders/: opendir failed: Not a directory
/bin/podman: stderr  stderr: dev/block/259:3/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/253:3/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/259:4/holders/: opendir failed: Not a directory
/bin/podman: stderr   dev/block/253:4/holders/: opendir failed: Not a directory
/bin/podman: stderr  stdout: Physical volume "/dev/nvme4n1" successfully created.
/bin/podman: stderr  stderr: Cannot archive volume group metadata for ceph-7bfd24e0-ec83-492d-977e-044d0d79f958 to read-only filesystem.
/bin/podman: stderr --> Was unable to complete a new OSD, will rollback changes
/bin/podman: stderr Running command: /usr/bin/ceph --cluster ceph --name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/ceph.ke>
/bin/podman: stderr  stderr: purged osd.0
/bin/podman: stderr -->  RuntimeError: command returned non-zero exit status: 5
```

**Comment by guits** [comment]
jenkins test ceph-volume tox

**Comment by guits** [comment]
jenkins test ceph-volume all

**Comment by guits** [comment]
jenkins test ceph-volume all

**Comment by guits** [comment]
jenkins test ceph-volume tox

**Comment by guits** [comment]
@sebastian-philipp could you retest it? Thanks!

**Comment by guits** [comment]
jenkins test make check

**Comment by guits** [comment]
https://pulpito.ceph.com/gabrioux-2021-11-10_07:46:10-rados:cephadm-wip-guits-testing-2021-11-09-1441-distro-default-smithi/

**Review by sebastian-philipp — APPROVED** [review]

**Pull request merged**